File: backend/src/main/java/vn/ClothingStore/domain/Review.java
package vn.ClothingStore.domain;

import java.time.Instant;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "review")
@Getter
@Setter
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int rating;
    @Column(columnDefinition = "MEDIUMTEXT")
    private String comment;
    private Instant createdAt;
    private Instant updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @OneToOne
    @JoinColumn(name = "order_detail_id")
    private OrderDetail orderDetail;

    @PrePersist
    public void handleBeforeCreate() {
        this.createdAt = Instant.now();
    }

    @PreUpdate
    public void handleBeforeUpdate() {
        this.updatedAt = Instant.now();
    }

}

File: frontend/src/app/dtos/review.ts
export interface ReviewDTO {
    id: number;
    rating: number;
    comment: string;
    createdAt: string;
    user: {
        id: number;
        name: string;
    };
    orderDetail: {
        id: number;
        nameProduct: string;
        color: string;
        size: string;
    };
}


export interface ReqReviewDTO {
    rating: number;
    comment: string;
    userId: number;
    productId: number;
    orderDetailId: number;
}

export interface ReqUpdateReviewDTO {
    rating: number;
    comment: string;
}